importlib_metadata/_py39compat.py

```diff
@@ -19,30 +19,17 @@ def normalized_name(dist: Distribution) -> Optional[str]:
     except AttributeError:
         from . import Prepared  # -> delay to prevent circular imports.
 
-        return Prepared.normalize(getattr(dist, "name", None) or dist.metadata['Name'])
+        return Prepared.normalize(dist.metadata['Name'])
```

Comment:
In newer versions of `importlib_metadata`, `_normalized_name` is resolved by normalizing `self.name`. Without inheritance, this is equivalent to `dist.metadata['Name']`. Inheritance, however, allows users to overwrite it.

Is there any chance this impacts here? (If there is, probably the risk is very low anyway, because old implementations would not expose the `.name` property, and newer implementations will not fail when accessing `dist._normalized_name`)

Reply:
That's a really good point. I do believe there are implementations out there that override `.name`, so I'll back out this change.